test/unit/storage/DataAccessorBasedStore.test.ts

```diff
@@ -604,5 +604,28 @@ describe('A DataAccessorBasedStore', (): void => {
       );
     });
   });
+
+  describe('resource Exists', (): void => {
+    it('should return false when the resource does not exist.', async(): Promise<void> => {
+      const resourceID = { path: `${root}resource` };
+      await expect(store.resourceExists(resourceID)).resolves.toBeFalsy();
+    });
+
+    it('should return true when the resource exists.', async(): Promise<void> => {
+      const resourceID = { path: `${root}resource` };
+      accessor.data[resourceID.path] = representation;
+      await expect(store.resourceExists(resourceID)).resolves.toBeTruthy();
+    });
+
+    it('should rethrow any unexpedted errors from validateIdentifier.', async(): Promise<void> => {
+      const resourceID = { path: `fail` };
+      const originalValidateIdentifier = store.validateIdentifier;
```

Comment:
Is this the reason `validateIdentifier` was made public? You could also test this by having the dataAccessor mock throw an error on `getMetadata` or the identifierStrategy that is being called in the `valideIdentifier` function.